e2e/nomostest/artifactregistry/crane.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package artifactregistry
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"kpt.dev/configsync/e2e/nomostest"
+)
+
+// SetupCraneImage creates a new CraneImage for use during an e2e test, along
+// with its registry, repository, and login.
+// Returns a reference to the Image object and any errors.
+func SetupCraneImage(nt *nomostest.NT, name, version string) (*CraneImage, error) {
+	image, err := SetupImage(nt, name, version)
+	if err != nil {
+		return nil, err
+	}
+	craneImage := &CraneImage{
+		Image: image,
+	}
+	if err := craneImage.RegistryLogin(); err != nil {
+		return nil, err
+	}
+	return craneImage, nil
+}
+
+// CraneImage represents a remote OCI-based crane-build image
+type CraneImage struct {
+	*Image
+}
+
+// RegistryLogin will log into the registry with crane using a gcloud auth token.
+func (r *CraneImage) RegistryLogin() error {
+	var err error
+	authCmd := r.Shell.Command("gcloud", "auth", "print-access-token")
+	loginCmd := r.Shell.Command("crane", "auth", "login",
```

Comment:
You chose to use `krane` in your other PR. What's the reason to switch to `crane`?

Reply:
That PR wasn't merged. @sdowell suggested another way to do it and made a different PR to replace it. So the current buildenv included crane but not krane.

Since our docs use crane, I stuck with that. The only think krane does is handle auth for you so you don't need to login or configure gcloud helper in the docker config.

Reply:
ACK